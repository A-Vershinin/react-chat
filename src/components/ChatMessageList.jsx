import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import ChatMessageItem from './ChatMessageItem.jsx';

/* eslint-disable */
const styles = theme => ({
  messagesWrapper: {
    overflowX: 'scroll',
    height: '100%',
    width: '100%',
    paddingTop: theme.spacing.unit * 8,
    paddingBottom: '120px',
  },
  paper: {
    padding: theme.spacing.unit * 3,
  },
});

const ChatMessageList = ({ classes, match, messages, activeUser }) => {
  if (!match.params.chatId) {
    return (
      <Paper className={classes.paper}>
        <Typography variant="display1" gutterBottom>
          Start messagingâ€¦
        </Typography>
        <Typography variant="body1" gutterBottom>
          Use <strong>Global</strong> to explore communities around here.
        </Typography>
        <Typography variant="body1" gutterBottom>
          Use <strong>Recents</strong> to see your recent conversations.
        </Typography>
      </Paper>
    );
  }

  return messages && messages.length ? (
    <div className={classes.messagesWrapper}>
      {messages &&
        messages.map((message, index) => {
          return <ChatMessageItem key={message._id} activeUser={activeUser} {...message} />;
        })}
    </div>
  ) : (
    <Typography variant="display1">There is no messages yet...</Typography>
  );
};

ChatMessageList.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  match: PropTypes.shape({
    params: PropTypes.object.isRequired,
  }).isRequired,
  activeUser: PropTypes.shape({
    firstName: PropTypes.string,
    lastName: PropTypes.string,
    username: PropTypes.string,
    isMember: PropTypes.bool.isRequired,
    isCreator: PropTypes.bool.isRequired,
    isChatMember: PropTypes.bool.isRequired,
  }).isRequired,
  messages: PropTypes.arrayOf(
    PropTypes.shape({
      chatId: PropTypes.string.isRequired,
      content: PropTypes.string.isRequired,
      sender: PropTypes.object.isRequired,
      createdAt: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default withRouter(withStyles(styles)(ChatMessageList));
