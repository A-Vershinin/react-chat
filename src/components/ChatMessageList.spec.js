/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ChatMessageList from './ChatMessageList.jsx';

jest.mock('./ChatMessageItem', () => () => 'ChatMessageItem');

const mockProps = {
  classes: {
    messagesWrapper: 'some text',
    paper: 'some text',
  },
  match: {
    params: {
      chatId: '12345',
    },
  },
  activeUser: {
    isMember: true,
    isCreator: true,
    isChatMember: true,
    firstName: 'firstName',
    lastName: 'lastName',
    username: 'username',
  },
  messages: [
    {
      chatId: '12345',
      content: 'some text',
      sender: {
        firstName: 'firstName',
        lastName: 'lastName',
        username: 'username',
        _id: '12345',
      },
      createdAt: '2018-03-16T10:53:23.200Z',
    },
  ],
};

describe('<ChatMessageList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ChatMessageList {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatMessageList {...mockProps} />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
