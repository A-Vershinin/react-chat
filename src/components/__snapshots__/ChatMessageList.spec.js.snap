// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMessageList /> renders correctly 1`] = `
<div
  className="MuiPaper-root-56 MuiPaper-elevation2-60 MuiPaper-rounded-57 ChatMessageList-paper-55 some text"
>
  <h1
    className="MuiTypography-root-83 MuiTypography-display1-87 MuiTypography-gutterBottom-100"
  >
    Start messagingâ€¦
  </h1>
  <p
    className="MuiTypography-root-83 MuiTypography-body1-92 MuiTypography-gutterBottom-100"
  >
    Use 
    <strong>
      Global
    </strong>
     to explore communities around here.
  </p>
  <p
    className="MuiTypography-root-83 MuiTypography-body1-92 MuiTypography-gutterBottom-100"
  >
    Use 
    <strong>
      Recents
    </strong>
     to see your recent conversations.
  </p>
</div>
`;
